creusot/tests/should_succeed/rusthorn/inc_max.mlcfg

```diff
@@ -65,23 +65,23 @@ module IncMax_TakeMax
   BB1 {
     assume { Resolve0.resolve mb };
     _9 <- Borrow.borrow_mut ( * ma);
-    ma <- { ma with current = ( ^ _9) };
+    ma <- Borrow.make_borrow ( ^ _9) ( ^ ma);
     _5 <- Borrow.borrow_mut ( * _9);
-    _9 <- { _9 with current = ( ^ _5) };
+    _9 <- Borrow.make_borrow ( ^ _5) ( ^ _9);
     assume { Resolve0.resolve _9 };
     goto BB3
   }
   BB2 {
     assume { Resolve0.resolve ma };
     _5 <- Borrow.borrow_mut ( * mb);
-    mb <- { mb with current = ( ^ _5) };
+    mb <- Borrow.make_borrow ( ^ _5) ( ^ mb);
     goto BB3
   }
   BB3 {
     _3 <- Borrow.borrow_mut ( * _5);
-    _5 <- { _5 with current = ( ^ _3) };
+    _5 <- Borrow.make_borrow ( ^ _3) ( ^ _5);
     _0 <- Borrow.borrow_mut ( * _3);
```

Comment:
I think this would mean losing something compared to the existing version of RustHornBelt, although I think this would be a necessary loss in order to allow equality in ghost functions. 

Specifically looking at the `inc_max` example it looks like mutable reference assignments are translated to use reborrowing instead of just being a why3 assignment which seems to be causing the issue.

Reply:
That's what the MIR is though, so that's the natural (compositional) translation.

Reply:
I wonder then if using the "third field" encoding would be better so we could keep " { _5 with current = ( ^ _3) }"

Reply:
Hmm, that makes things difficult. If we made `borrowed` a struct we 3 fields we could get back lines like `{ _5 with current = ( ^ _3) }`, but this would help prove this example since we are returning `Borrow.borrow_mut ...` which would make no guarantees about the third field. If we also special-cased `Borrow.borrow_mut` when applied to the current value of a mutable reference to guarantee the third field matched, I think we could prove the example. However, I think that this would reintroduce the equality unsoundness since it could be written like:
```rust
#[ensures(false)]
pub fn bad() {
    let mut x = gh! { true };
    let mut xm = &mut x;
    let b = &mut *xm;
    let bg = gh! { b };
    proof_assert! { ***bg == true && *^*bg == true };
    let evil = &mut *xm;
    proof_assert! { (evil == *bg) == (*^evil == true) }; // Since **evil == true
    *evil = gh! { if evil == *bg { false } else { true } };
    proof_assert! { **evil == !*^evil };
    // Resolve evil
    proof_assert! { **evil == !**evil }; //BOOM !
}
```